common/lib/capa/capa/capa_problem.py

```diff
@@ -789,7 +793,9 @@ def _extract_html(self, problemtree):  # private
             answervariable = None
             if problemid in self.correct_map:
                 pid = input_id
-                status = self.correct_map.get_correctness(pid)
+                if not self.saved_answers:
```

Comment:
I wonder if here you should show the correctness if the saved answer matches the submitted answer.